### PR TITLE
feat: include kill count for npc loot notifications

### LINKED ISSUE #296: Is it possible to also include the KC of when a drop happens?
The plugin works great but I was hoping there would be a way to also include the KC for the drop as well. I'm not too familiar with how the code works, but it should be at least possible to include the bosses' KC. Non boss npc kills, pickpockets, and chest openings, etc. might be a little more complicated.

Again, I'm not very knowledgeable about how the code works, but if this seems possible based on what is available from other plugins then I'm more than happy to help it whatever way I can.

### PR DESCRIPTION
**Pull request by iProdigy**
Closes #296 (Related https://github.com/RinZJ/better-discord-loot-logger/issues/22)

image

This PR relies upon the base loot tracker plugin observing all kills. In a future PR, we will use boss chat messages to override base loot tracker for more accurate kc stats

### CODE REVIEW

**Comment by Felanbird** [comment]
I think the main issue is coming from the KC value not correctly getting updated, or we're not reading it correctly.
When killing enemies in quick succession, the KC is not updated, but after x amount of time the KC does update (checked quicker-succession and that didn't work, likely real-time). 
First kill and update kill being 1-off always tho.

<br/><br/>

Red: KC as reported by dink
Lime Green: Correct KC
Cyan: Timestamp

<details>
<summary>Rune dragons:</summary>

image


I disabled JSON after the 4 kills since I figured it wasn't going to change, which is why some timestamps have seconds
```
   "extra":{
      "items":[
         {
            "id":2363,
            "quantity":1,
            "priceEach":12280,
            "name":"Runite bar"
         },
         {
            "id":19580,
            "quantity":26,
            "priceEach":515,
            "name":"Rune javelin heads"
         }
      ],
      "source":"Rune dragon",
      "category":"NPC",
      "killCount":571
   },
```
```
   "extra":{
      "items":[
         {
            "id":2363,
            "quantity":1,
            "priceEach":12280,
            "name":"Runite bar"
         },
         {
            "id":19582,
            "quantity":31,
            "priceEach":889,
            "name":"Dragon javelin heads"
         }
      ],
      "source":"Rune dragon",
      "category":"NPC",
      "killCount":571
   },
```
```
   "extra":{
      "items":[
         {
            "id":2363,
            "quantity":1,
            "priceEach":12280,
            "name":"Runite bar"
         },
         {
            "id":562,
            "quantity":88,
            "priceEach":45,
            "name":"Chaos rune"
         }
      ],
      "source":"Rune dragon",
      "category":"NPC",
      "killCount":571
   },
```
```
   "extra":{
      "items":[
         {
            "id":2363,
            "quantity":1,
            "priceEach":12280,
            "name":"Runite bar"
         },
         {
            "id":1303,
            "quantity":1,
            "priceEach":18408,
            "name":"Rune longsword"
         }
      ],
      "source":"Rune dragon",
      "category":"NPC",
      "killCount":571
   },
```
</details>

<details>
<summary>H.A.M Guards</summary>

image

</details>

File: src/main/java/dinkplugin/util/SerializedLoot.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package dinkplugin.util;
+
+import lombok.AccessLevel;
+import lombok.Data;
+import lombok.Setter;
+
+@Data
+@Setter(AccessLevel.PRIVATE)
+public class SerializedLoot {
```
Comment (pajlada) [inline src/main/java/dinkplugin/util/SerializedLoot.java:9]: Comment specifying where this comes from, if it's a struct from Runelite. maybe link to the file with a github link?

**Review by pajlada — CHANGES_REQUESTED** [review]
Looks like a very reasonable implementation :+1: Small nitpicks/comments

**Review by pajlada — APPROVED** [review]
Code looks good to me, and I agree with the multi-step approach of this feature.
Basic Loot Tracker-based KC first, more advanced KC-tracking later (bosses etc)

**Review by Felanbird — APPROVED** [review]
Seems to now work as I would expect it to 👍

**Pull request merged**